packages/ember-data/lib/system/relationships/has_many.js

```diff
@@ -104,10 +104,14 @@ function hasMany(type, options) {
     key: null
   };
 
-  return Ember.computed(function(key) {
+  return Ember.computed(function(key, records) {
     var relationship = this._relationships[key];
+    if (arguments.length === 2) {
+      relationship.removeRecords(relationship.getRecords());
```

Comment:
When using `async: true` I think this line would kick off a server request, no? We probably don't want that when we do:

```
store.createRecord('parent', { children: [child1, child2] });
```